app/fastlane/helpers/slack.rb

```
@@ -0,0 +1,69 @@
+          complete_response = complete_http.request(complete_request)
+          raise "Slack API failed: #{complete_response.code} #{complete_response.body}" unless complete_response.is_a?(Net::HTTPSuccess)
+          json = JSON.parse(complete_response.body)
+          raise "Slack API Error: #{json["error"]}" unless json["ok"]
+          final_info = json["files"]&.first
+        end
+        report_success("Successfully uploaded and shared #{file_name} to Slack channel #{channel_id}")
+        final_info
+      rescue => e
+        report_error("Error during Slack upload process: #{e.message}", e.backtrace.join("\n"), "Slack Upload Failed")
+        false
+      end
```

RuboCop (1.75.5)

[convention] Assignment Branch Condition size for `upload_file_to_slack` is too high. [<38, 77, 12> 86.7/23]

(Metrics/AbcSize)